Makefile

```diff
@@ -87,24 +96,14 @@ e2e-janitor:
 .PHONY: docker-build
 docker-build: generate lint-full ## Build the docker image for controller-manager
 	docker build --pull . -t $(MANAGER_IMAGE)
+	# TODO: sed probably needs to be gnu sed not bsd sed
+	sed -i '' -e 's@image: .*@image: '"${MANAGER_IMAGE}"'@' ./config/default/manager_image_patch.yaml
+	sed -i '' -e 's@credentials: .*@credentials: '"${B64_CREDENTIALS}"'@' ./config/default/aws_credentials_patch.yaml
```

Comment:
Should this be part of the `docker-build` target? I would expect it as part of `generate-manifests` instead.

Reply:
This is how kubebuilder v2 makefiles are.

I believe the idea behind this being part of the docker-build step is that you'd build for a release/deployment. At which point the sed is applied to the image patch and you can now deploy your app.

If it were part of the generate-manifest we would only be guessing at the name of the image. So it might look correct but it wouldn't be valid until the docker-build/push step anyway

Reply:
can't say I'm a fan, but we can go with it for now and reassess it later.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1014